homeassistant/components/vicare/sensor.py

```diff
@@ -597,7 +597,7 @@ async def _entities_from_descriptions(
     hass: HomeAssistant,
     entities: list[ViCareSensor],
     sensor_descriptions: tuple[ViCareSensorEntityDescription, ...],
-    iterables,
+    iterables: list[PyViCareDevice],
```

Comment:
Same as above, use `HeatingDeviceWithComponent`

```suggestion
    iterables: list[HeatingDeviceWithComponent],
```